cmake/Modules/CocosBuildSet.cmake

```diff
@@ -1,11 +1,3 @@
-# Disable in-source builds to prevent source tree corruption.
-if(" ${CMAKE_SOURCE_DIR}" STREQUAL " ${CMAKE_BINARY_DIR}")
```

Comment:
suggest to keep this check, it will prevent In-source builds

Reply:
I think that we should not force on user that. 
I propose to keep cmake files as simple as possible.